### PR TITLE
Add repo options to yum package module

### PR DESCRIPTION
**Pull request by mikeweilgart**
This replaces and improves upon the un-accepted pull request cfengine/masterfiles#809.

This PR adds support for enablerepo and disablerepo flags to be passed via options
to work for list-updates, list-updates-local, and repo-install.

### CODE REVIEW

**Comment by mikeweilgart** [comment]
@nickanderson

**Comment by nickanderson** [comment]
@kacf @olehermanse  can you please review?

**Comment by nickanderson** [comment]
@mikeweilgart  we will also need the documentation for the module updated to reflect support for `enablerepo` and `disablerepo` options. https://github.com/cfengine/documentation/blob/master/reference/promise-types/packages.markdown#yum

File: modules/packages/yum [hunk 1]
```diff
@@ -103,8 +103,12 @@ def list_installed():
 
 
 def list_updates(online):
-    # Ignore everything.
-    sys.stdin.readlines()
+    for line in sys.stdin:
+        if line.startswith("options="):
+            option = line[8:]
```
Comment (olehermanse) [inline modules/packages/yum]:
```
option = line[len("options="):]
```

File: modules/packages/yum [hunk 2]
```diff
@@ -198,6 +202,11 @@ def package_arguments_builder(is_yum_install):
     multi_cmd_args = []     # List of lists of arguments
     old_name = ""
     for line in sys.stdin:
+        if line.startswith("options="):
+            option = line[8:]
```
Comment (olehermanse) [inline modules/packages/yum]: Same as above

File: modules/packages/yum (hunk 1 above)
Comment (olehermanse) [reply modules/packages/yum]: Also, does it make sense to trim some whitespace from each line? i.e.:
```
line = line.strip()
```

**Comment by mikeweilgart** [comment]
@olehermanse, done.  Can you please review again?

**Comment by olehermanse** [comment]
@mikeweilgart Looks good! Please squash your commits, and @nickanderson or I will merge it.

**Comment by mikeweilgart** [comment]
@olehermanse squashed!  Have you had a chance to test this in action yet?  (I don't recall exactly what testing I did, but I believe I tested only at the package module API level, not end-to-end integration with CFEngine policy.)

**Comment by olehermanse** [comment]
@mikeweilgart I will test this on a CentOS VM tomorrow

File: modules/packages/yum [hunk 3]
```diff
@@ -103,8 +103,13 @@ def list_installed():
...
 def list_updates(online):
-    # Ignore everything.
-    sys.stdin.readlines()
+    for line in sys.stdin:
+        line = line.strip()
+        if line.startswith("options="):
+            option = line[len("options="):]
+            if option.startswith("enablerepo=") or option.startswith("disablerepo="):
+                global yum_options
+                yum_options += ["--" + option]
```
Comment (vpodzime) [inline modules/packages/yum]: There's no need to use ``global`` when modifying an existing object. Also, ``yum_options.append()`` would be less cryptic here, I think. And the same of course applies below too.

**Comment by olehermanse** [comment]
I just tested this on a CentOS machine:
```
body file control
{
  inputs => { "$(sys.libdir)/stdlib.cf" };
}

bundle agent main
{
  packages:
    "git"
        policy => "present",
        package_module => yum,
        options => { "enablerepo=updates" };
}
```
With different combinations of enablerepo, disablerepo, updates and base. Works as expected.

Thanks @mikeweilgart, merging.

**Comment by vpodzime** [comment]
@nickanderson could you please merge this as long as it's all set? (don't know about the docs update)

**Comment by nickanderson** [comment]
Docs haven't been written for it yet. I'd prefer to hold this until that gets done.

**Comment by nickanderson** [comment]
This is where docs need to be added: https://github.com/cfengine/documentation/blob/master/reference/promise-types/packages.markdown#yum

**Comment by vpodzime** [comment]
@mikeweilgart, could you please submit a PR for the docs too?

**Comment by nickanderson** [comment]
Thanks for the contribution!

Merging https://github.com/cfengine/masterfiles/pull/1069 with empty changelog commit.

**Pull request closed without merging**